include/ensmallen_bits/cne/cne_impl.hpp

```diff
@@ -104,7 +104,7 @@ typename MatType::elem_type CNE::Optimize(ArbitraryFunctionType& function,
       << std::endl;
 
   // Find the fitness before optimization using given iterate parameters.
-  ElemType lastBestFitness = function.Evaluate(iterate);
+  size_t lastBestFitness = function.Evaluate(iterate);
```

Comment:
I think using ``ElemType``makes sense, the main issue is that the optimisation process stops if the current and previous fitness/loss is the same:

https://github.com/mlpack/ensmallen/blob/9d2a5f07c63e292d116c5b182a7682615798f04e/include/ensmallen_bits/cne/cne_impl.hpp#L108

Depending on the problem this is more or less likely. As a workaround I used ``-1`` as threshold in https://github.com/mlpack/ensmallen/pull/119/commits/2eda0cb86ba661e7072c27795581a89da17209f6, if we do this we can use a much lower population size and number of iterations as well.

Reply:
Ah, ok.  Since that commit is in #119, should we merge this as-is despite the failing build and then merge #119 just afterwards?

Reply:
Are you talking about merging #113 as-is? Not sure, have to double check, but I think #119 does contain every commit from this PR, so I guess, it would be fine to just merge #119, once it is done (the ``StepTake`` additions is almost done)?

Reply:
Oh, I see---yeah, we should just merge #119 all at once then.  Maybe I made your life a little hard with the merge 3c72dae though; let me know if you want me to reproduce that merge on the `callbacks` branch (and cherry-pick a8a2ff0).

Reply:
The last time I just used cherry-pick on your branch, so I think I can do the same for the latest commits, but let's see.